evm/evm.go

```diff
@@ -144,12 +140,13 @@ type Evm struct {
 	chainConfig     params.ChainConfig
 	vmConfig        vm.Config
 	validateTxValue bool
+	gasLimit        uint64
 }
 
 func NewEvm(sdb vm.StateDB, lstate loomchain.State, abm *evmAccountBalanceManager, debug bool) *Evm {
 	p := new(Evm)
 	p.sdb = sdb
-
+	p.gasLimit = lstate.Config().GetEvm().GasLimit
```

Comment:
Suppose you set `Config.Evm.GasLimit` to zero, then the protobuf unmarshaller will unmarshal `Config.Evm` to `nil`, and then I suspect `lstate.Config().GetEvm().GasLimit` will cause a panic due to `nil` being de-referenced... might be a good idea to write a small test.